### DISCUSSION TITLE
Please give your feedback to LinkAce 2 features

### DISCUSSION THREAD

**Discussion by Kovah** [Kovah/LinkAce#555] [Announcements]
Dear LinkAce users,

I am working on version 2 of the app from time to time, but It becomes increasingly difficult for me to make and justify any decisions regarding all the new features or changes that are planned.

The major problem is that most of those features are really big and require a lot of work at the core. For example, support for multiple users has let to multiple big stops because getting permissions and the user experience right is _really_ hard. Just read through the ticket and you might understand what I am going through. 😅
The same goes for SSO and LDAP, which I haven't even started working on. Last but not least, I am not a designer or UX expert, so any work in the LinkAce interface is more trial and error than actual work.

To sum this up: **I need your help.** I need feedback on all those features that are planned. I need the opinion of those using LinkAce, and hopefully help from those who have had the same issues with their applications. Otherwise, if you think you know someone who can help, please forward this.

### 👉  **[LinkAce 2 features](https://github.com/Kovah/LinkAce/milestone/4)**

Thanks,
Kevin

**Comment by habitats-tech** [comment]
For me, multiuser support and additional authentication methods should be a paid for option only, because this is usually asked by people who work for organisations and should pay for your effort as they can afford it.

I would like to suggest a couple of enhancements, which should not be too taxing on development effort.

1. Badges: We have Lists and Tags. I suggest adding one more, named Badges, to allow for emoticons to be used, which could provide a 3rd level of classification.

2. Links: the ability to decide, through settings, if badges, lists, tags, are displayed. Right now under Links there is no ability to see on which lists a link belongs to.

3. Links: Ability to filter by Badges, Lists, Tags.

**Reply by Kovah** [reply]
I have already thought about making enterprise features such as SSO or LDAP authentication a paid feature that one can unlock by buying a license. However, just the legal aspects alone makes it hard for me as a single developer to consider this option any further. And I won't hire a lawyer to discuss something like this.
Your idea about setting up some sort of fund for this is an idea worth exploring. However, it is always difficult for me to find the correct balance between LinkAce, my other projects and my actual day-job. I don't want to stress myself out because I promised to work on topics like SSO when businesses pay money. Because.. actually it's not really about money. It's about time.

I totally understand your idea and concerns, it's just not that easy. Sadly.

**Reply by Kovah** [reply]
I decided that LinkAce will not get support for LDAP / SSO yet. I am searching for a company willing to sponsor this feature. More details: https://github.com/Kovah/LinkAce/discussions/671

**Comment by habitats-tech** [comment]
I suggest you create a voluntary contribution monetary fund for anyone who wishes to see enhanced authentication and multiuser support.

**Comment by habitats-tech** [comment]
IMHO you could ask for help from the community sharing the contributions people, who want enterprise features, make. This way everyone wins and the development effort/burden is shared, plus you get some quality time with your family. LinkAce could become a killer product if you keep it simple and get a handful of good skilful people involved. As you know all it takes is finding people who share your skill, passion and vision.

**Comment by cweiske** [comment]
SSO, LDAP and OpenID Connect sound like enterprise features that are not needed by "normal" self-hosted users. Let people contribute when they want those features.

I personally do not see why a big "2.0" is needed when gradual improvements would work as well.

**Comment by Christea-12** [comment]
Sorry, I can't find an effective feedback method, so I can only leave a message under this post. This paragraph was machine translated, so please forgive me if there are any errors.
I'd like to make some suggestions as I use Linkace as my link management app. But I encountered a problem, that is, I cannot manage and delete links in batches.
Therefore I hope you can add a select or select all button in the next version so that I can manage these links in batches.
In addition, I have an idea, because you provide the option of managing by tags and managing by folders. In fact, we can also set it so that if a tag is deleted, then it will be deleted in batches, including the content of the tag, or if the folder is deleted, Batch delete the contents of containing folders.
Bulk deletion of links by folder or tag can be turned on or off, making it easier to manage large numbers of bookmarks.
